make_profiler/report/script.js

```diff
@@ -0,0 +1,153 @@
+function filterTarget() {
+    let input, filter, table, tr, td, i, txtValue;
+    input = document.getElementById("myInput");
+
+    filter = input.value.toUpperCase().split(" ").join("_");
+    table = document.getElementById("statusTable");
+    tr = table.getElementsByTagName("tr");
+
+    if (filter.startsWith('"') && filter.endsWith('"')) {
+        filter = filter.substring(1, filter.length - 1)
+        for (i = 0; i < tr.length; i++) {
+            td = tr[i].getElementsByTagName("td")[0];
+            if (td) {
+                txtValue = td.childNodes[0].textContent || td.innerText;
+                if (txtValue.toUpperCase() === filter) {
+                    tr[i].style.display = "";
+                } else {
+                    tr[i].style.display = "none";
+                }
+            }
+        }
+    } else {
+        for (i = 0; i < tr.length; i++) {
+            td = tr[i].getElementsByTagName("td")[0];
+            if (td) {
+                txtValue = td.textContent || td.innerText;
+                if (txtValue.toUpperCase().includes(filter)) {
+                    tr[i].style.display = "";
+                } else {
+                    tr[i].style.display = "none";
+                }
+            }
+        }
+    }
+}
```

Comment:
I think I didn't miss anything but still needs checking
```suggestion
function filterTarget() {
    // Comment about what this function does (since it returns nothing and transforms DOM (html))
    let filtered, txtValue;
    const inputElement = document.getElementById("myInput")
    const inputValue = inputElement?.value;
    if (typeof inputValue !== 'string') console.error('no input value provided for', inputValue)

    filtered = inputValue.toUpperCase().split(" ").join("_");
    const tableElement = document.getElementById("statusTable");
    if (!tableElement) console.error('table was not found')
    const tableRows = tableElement.getElementsByTagName("tr");

    for (let i = 0; i < tableRows.length; i++) {
        const firstTD = tableRows[i].getElementsByTagName("td")[0]
        let wasWrappedInQuotationMarks = false

        if (filtered.startsWith('"') && filtered.endsWith('"')) {
            filtered = filtered.substring(1, filtered.length - 1)
            wasWrappedInQuotationMarks = true
        }

        // Comment about why we treat those cases differently
        if (firstTD && wasWrappedInQuotationMarks) {
            txtValue = firstTD.childNodes[0].textContent || firstTD.innerText;
            if (txtValue.toUpperCase() === filtered) {
                tableRows[i].style.display = "unset";
            } else {
                tableRows[i].style.display = "none";
            }
        } else if (firstTD) {
            txtValue = firstTD.textContent || firstTD.innerText;
            if (txtValue.toUpperCase().includes(filtered)) {
                tableRows[i].style.display = "unset";
            } else {
                tableRows[i].style.display = "none";
            }
        }
    }
}
```

Reply:
no way :(